packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -246,15 +246,11 @@ class CupertinoDatePicker extends StatefulWidget {
     this.use24hFormat = false,
     this.backgroundColor,
   }) : initialDateTime = initialDateTime ?? DateTime.now(),
-       assert(mode != null),
```

Comment:
We're not yet removing these asserts, since it is still possible to run Flutter in non-NNBD mode. We'll do a pass later and remove all of these at once (there's a lint that we'll turn on).